### ISSUE TITLE
[BUG] External Kubeconfig cloud credentials don't work with node driver

### ISSUE THREAD

**Issue by noahgildersleeve** [harvester/harvester#2220]
**Describe the bug**

If you select an external kubeconfig for your cloud credentials in Rancher it won't successfully provision RKE1 or RKE2 clusters in node driver
**To Reproduce**
1. Import Harvester into Rancher
2. Get Kubeconfig from support section of Harvester
3. Create cloud credentials with external Harvester
4. Create RKE1 or RKE2 cluster with manually entered details.

**Expected behavior**

The cluster should provision
**Support bundle**

[supportbundle_104089c1-1eb0-475e-bf5c-f4cc35c4ea65_2022-05-05T00-28-39Z.zip](https://github.com/harvester/harvester/files/8628187/supportbundle_104089c1-1eb0-475e-bf5c-f4cc35c4ea65_2022-05-05T00-28-39Z.zip)

**Environment:**
 - Harvester ISO version: v1.0.2-rc1
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): vagrant 3 node setup
 - Rancher Version: v2.6.5-rc6
 
**Additional context**
I tested this with RKE1, RKE2 on v1.23 and both with v1.22. This was with a 1 node Docker install on a separate VM. There were no DNS entries used in the config and only IPs. Imported Docker worked fine.

```
 Trying to access option  which does not exist
 THIS ***WILL*** CAUSE UNEXPECTED BEHAVIOR
 Type assertion did not go smoothly to string for key 
 Running pre-create checks...
 Creating machine...
 (rke2-ext-final-opensuse-rke2-ext-final-36d034c5-jkhxg) Creating New SSH Key
 (rke2-ext-final-opensuse-rke2-ext-final-36d034c5-jkhxg) Using public Key: /home/machine/.docker/machine/machines/rke2-ext-final-opensuse-rke2-ext-final-36d034c5-jkhxg/id_rsa.pub
 (rke2-ext-final-opensuse-rke2-ext-final-36d034c5-jkhxg) Start node
 (rke2-ext-final-opensuse-rke2-ext-final-36d034c5-jkhxg) Waiting for node become Running
 Error creating machine: Error in driver during machine creation: Too many retries waiting for machine to be Running.  Last error: Maximum number of retries (120) exceeded
 The default lines below are for a sh/bash shell, you can specify the shell you're using, with the --shell flag., waiting for agent to check in and apply initial plan
 2022/05/05 00:18:43 [INFO] [planner] rkecluster fleet-default/rke2-ext-final: waiting: configuring bootstrap node(s) rke2-ext-final-opensuse-rke2-ext-final-5bf455cf66-s6fxv: failed creating server [fleet-default/rke2-ext-final-opensuse-rke2-ext-final-36d034c5-jkhxg] of kind (HarvesterMachine) for machine rke2-ext-final-opensuse-rke2-ext-final-5bf455cf66-s6fxv in infrastructure provider: CreateError: Downloading driver from https://192.168.0.52/assets/docker-machine-driver-harvester
 Doing /etc/rancher/ssl
 docker-machine-driver-harvester
 docker-machine-driver-harvester: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), statically linked, stripped
 Trying to access option  which does not exist
 THIS ***WILL*** CAUSE UNEXPECTED BEHAVIOR
 Type assertion did not go smoothly to string for key 
 Running pre-create checks...
 Creating machine...
 (rke2-ext-final-opensuse-rke2-ext-final-36d034c5-jkhxg) Creating New SSH Key
 (rke2-ext-final-opensuse-rke2-ext-final-36d034c5-jkhxg) Using public Key: /home/machine/.docker/machine/machines/rke2-ext-final-opensuse-rke2-ext-final-36d034c5-jkhxg/id_rsa.pub
 (rke2-ext-final-opensuse-rke2-ext-final-36d034c5-jkhxg) Start node
 (rke2-ext-final-opensuse-rke2-ext-final-36d034c5-jkhxg) Waiting for node become Running
 Error creating machine: Error in driver during machine creation: Too many retries waiting for machine to be Running.  Last error: Maximum number of retries (120) exceeded
 The default lines below are for a sh/bash shell, you can specify the shell you're using, with the --shell flag., waiting for agent to check in and apply initial plan
 2022/05/05 00:18:43 [INFO] [planner] rkecluster fleet-default/rke2-ext-final: waiting: configuring bootstrap node(s) rke2-ext-final-opensuse-rke2-ext-final-5bf455cf66-s6fxv: failed creating server [fleet-default/rke2-ext-final-opensuse-rke2-ext-final-36d034c5-jkhxg] of kind (HarvesterMachine) for machine rke2-ext-final-opensuse-rke2-ext-final-5bf455cf66-s6fxv in infrastructure provider: CreateError: Downloading driver from https://192.168.0.52/assets/docker-machine-driver-harvester
 Doing /etc/rancher/ssl
 docker-machine-driver-harvester
 docker-machine-driver-harvester: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), statically linked, stripped
 Trying to access option  which does not exist
 THIS ***WILL*** CAUSE UNEXPECTED BEHAVIOR
 Type assertion did not go smoothly to string for key 
 Running pre-create checks...
 Creating machine...
 (rke2-ext-final-opensuse-rke2-ext-final-36d034c5-jkhxg) Creating New SSH Key
 (rke2-ext-final-opensuse-rke2-ext-final-36d034c5-jkhxg) Using public Key: /home/machine/.docker/machine/machines/rke2-ext-final-opensuse-rke2-ext-final-36d034c5-jkhxg/id_rsa.pub
 (rke2-ext-final-opensuse-rke2-ext-final-36d034c5-jkhxg) Start node
 (rke2-ext-final-opensuse-rke2-ext-final-36d034c5-jkhxg) Waiting for node become Running
 Error creating machine: Error in driver during machine creation: Too many retries waiting for machine to be Running.  Last error: Maximum number of retries (120) exceeded
 The default lines below are for a sh/bash shell, you can specify the shell you're using, with the --shell flag., waiting for agent to check in and apply initial plan
 2022/05/05 00:18:43 [INFO] [MachineProvision] Failed to create infrastructure fleet-default/rke2-ext-final-opensuse-rke2-ext-final-36d034c5-jkhxg for machine rke2-ext-final-opensuse-rke2-ext-final-5bf455cf66-s6fxv, deleting and recreating...
 2022/05/05 00:18:43 [ERROR] [planner] rkecluster fleet-default/rke2-ext-final: error encountered during plan processing was init node not found
 2022/05/05 00:18:44 [ERROR] [planner] rkecluster fleet-default/rke2-ext-final: error encountered during plan processing was init node not found
 2022/05/05 00:18:44 [ERROR] [planner] rkecluster fleet-default/rke2-ext-final: error encountered during plan processing was init node not found
 2022/05/05 00:18:44 [INFO] [MachineProvision] Failed to create infrastructure fleet-default/rke2-ext-final-opensuse-rke2-ext-final-36d034c5-jkhxg for machine rke2-ext-final-opensuse-rke2-ext-final-5bf455cf66-s6fxv, deleting and recreating...
 2022/05/05 00:18:44 [ERROR] [planner] rkecluster fleet-default/rke2-ext-final: error encountered during plan processing was init node not found
 2022/05/05 00:18:44 [ERROR] [planner] rkecluster fleet-default/rke2-ext-final: error encountered during plan processing was init node not found
 2022/05/05 00:18:44 [ERROR] [planner] rkecluster fleet-default/rke2-ext-final: error encountered during plan processing was init node not found
 2022/05/05 00:18:44 [ERROR] [planner] rkecluster fleet-default/rke2-ext-final: error encountered during plan processing was init node not found
 2022/05/05 00:18:49 [ERROR] [planner] rkecluster fleet-default/rke2-ext-final: error encountered during plan processing was init node not found
 2022/05/05 00:18:52 [ERROR] error syncing '_all_': handler user-controllers-controller: failed to start user controllers for cluster c-m-2dbqq5sw: ClusterUnavailable 503: cluster not found, requeuing
 2022/05/05 00:18:54 [ERROR] [planner] rkecluster fleet-default/rke2-ext-final: error encountered during plan processing was init node not found
 2022/05/05 00:18:59 [ERROR] [planner] rkecluster fleet-default/rke2-ext-final: error encountered during plan processing was init node not found
 2022/05/05 00:18:59 [ERROR] [planner] rkecluster fleet-default/rke2-ext-final: error encountered during plan processing was init node not found
 2022/05/05 00:18:59 [ERROR] [planner] rkecluster fleet-default/rke2-ext-final: error encountered during plan processing was init node not found
 2022/05/05 00:18:59 [ERROR] error syncing 'fleet-default/rke2-ext-final-bootstrap-template-7bzs9': handler rke-machine: failed to delete fleet-default/rke2-ext-final-bootstrap-template-7bzs9-machine-plan /v1, Kind=Secret for rke-machine fleet-default/rke2-ext-final-bootstrap-template-7bzs9: secrets "rke2-ext-final-bootstrap-template-7bzs9-machine-plan" not found, requeuing
 2022/05/05 00:18:59 [ERROR] error syncing 'fleet-default/rke2-ext-final-bootstrap-template-7bzs9': handler rke-machine: failed to delete fleet-default/rke2-ext-final-bootstrap-template-7bzs9-machine-plan /v1, Kind=Secret for rke-machine fleet-default/rke2-ext-final-bootstrap-template-7bzs9: secrets "rke2-ext-final-bootstrap-template-7bzs9-machine-plan" not found, requeuing
 2022/05/05 00:18:59 [INFO] [planner] rkecluster fleet-default/rke2-ext-final: waiting: waiting for at least one bootstrap node
 2022/05/05 00:18:59 [INFO] [planner] rkecluster fleet-default/rke2-ext-final: waiting: waiting for at least one bootstrap node
 2022/05/05 00:18:59 [ERROR] Reconciler error: machines.cluster.x-k8s.io "rke2-ext-final-opensuse-rke2-ext-final-5bf455cf66-s6fxv" not found
 2022/05/05 00:18:59 [INFO] [planner] rkecluster fleet-default/rke2-ext-final: waiting: waiting for at least one bootstrap node
 2022/05/05 00:19:00 [INFO] [planner] rkecluster fleet-default/rke2-ext-final: waiting: waiting for at least one bootstrap node
 2022/05/05 00:19:00 [INFO] [planner] rkecluster fleet-default/rke2-ext-final: waiting: waiting for at least one bootstrap node
 2022/05/05 00:19:00 [ERROR] error syncing 'fleet-default/rke2-ext-final-opensuse-rke2-ext-final-36d034c5-f8bq8': handler machine-provision: no machine owner ref, requeuing
 2022/05/05 00:19:00 [INFO] [planner] rkecluster fleet-default/rke2-ext-final: waiting: waiting for at least one bootstrap node
 2022/05/05 00:19:00 [ERROR] [planner] rkecluster fleet-default/rke2-ext-final: error encountered during plan processing was Operation cannot be fulfilled on secrets "rke2-ext-final-bootstrap-template-vxztd-machine-plan": the object has been modified; please apply your changes to the latest version and try again
 2022/05/05 00:19:00 [INFO] [planner] rkecluster fleet-default/rke2-ext-final: waiting: configuring bootstrap node(s) rke2-ext-final-opensuse-rke2-ext-final-5bf455cf66-f4vzk: waiting for bootstrap etcd to be available
 2022/05/05 00:19:00 [INFO] [planner] rkecluster fleet-default/rke2-ext-final: waiting: configuring bootstrap node(s) rke2-ext-final-opensuse-rke2-ext-final-5bf455cf66-f4vzk: waiting for agent to check in and apply initial plan
 2022/05/05 00:19:00 [INFO] [planner] rkecluster fleet-default/rke2-ext-final: waiting: configuring bootstrap node(s) rke2-ext-final-opensuse-rke2-ext-final-5bf455cf66-f4vzk: waiting for agent to check in and apply initial plan
 2022/05/05 00:19:01 [INFO] [planner] rkecluster fleet-default/rke2-ext-final: waiting: configuring bootstrap node(s) rke2-ext-final-opensuse-rke2-ext-final-5bf455cf66-f4vzk: waiting for agent to check in and apply initial plan
 2022/05/05 00:19:01 [INFO] [planner] rkecluster fleet-default/rke2-ext-final: waiting: configuring bootstrap node(s) rke2-ext-final-opensuse-rke2-ext-final-5bf455cf66-f4vzk: waiting for agent to check in and apply initial plan
 2022/05/05 00:19:01 [INFO] [planner] rkecluster fleet-default/rke2-ext-final: waiting: configuring bootstrap node(s) rke2-ext-final-opensuse-rke2-ext-final-5bf455cf66-f4vzk: waiting for agent to check in and apply initial plan
 2022/05/05 00:19:01 [INFO] [planner] rkecluster fleet-default/rke2-ext-final: waiting: configuring bootstrap node(s) rke2-ext-final-opensuse-rke2-ext-final-5bf455cf66-f4vzk: waiting for agent to check in and apply initial plan
 2022/05/05 00:19:01 [INFO] [planner] rkecluster fleet-default/rke2-ext-final: waiting: configuring bootstrap node(s) rke2-ext-final-opensuse-rke2-ext-final-5bf455cf66-f4vzk: waiting for agent to check in and apply initial plan
 2022/05/05 00:19:01 [INFO] [planner] rkecluster fleet-default/rke2-ext-final: waiting: configuring bootstrap node(s) rke2-ext-final-opensuse-rke2-ext-final-5bf455cf66-f4vzk: waiting for agent to check in and apply initial plan
 2022/05/05 00:19:01 [INFO] [planner] rkecluster fleet-default/rke2-ext-final: waiting: configuring bootstrap node(s) rke2-ext-final-opensuse-rke2-ext-final-5bf455cf66-f4vzk: waiting for agent to check in and apply initial plan
 2022/05/05 00:19:02 [INFO] [planner] rkecluster fleet-default/rke2-ext-final: waiting: configuring bootstrap node(s) rke2-ext-final-opensuse-rke2-ext-final-5bf455cf66-f4vzk: creating server [fleet-default/rke2-ext-final-opensuse-rke2-ext-final-36d034c5-f8bq8] of kind (HarvesterMachine) for machine rke2-ext-final-opensuse-rke2-ext-final-5bf455cf66-f4vzk in infrastructure provider, waiting for agent to check in and apply initial plan
 2022/05/05 00:19:02 [INFO] [planner] rkecluster fleet-default/rke2-ext-final: waiting: configuring bootstrap node(s) rke2-ext-final-opensuse-rke2-ext-final-5bf455cf66-f4vzk: creating server [fleet-default/rke2-ext-final-opensuse-rke2-ext-final-36d034c5-f8bq8] of kind (HarvesterMachine) for machine rke2-ext-final-opensuse-rke2-ext-final-5bf455cf66-f4vzk in infrastructure provider, waiting for agent to check in and apply initial plan
 2022/05/05 00:19:03 [INFO] [planner] rkecluster fleet-default/rke2-ext-final: waiting: configuring bootstrap node(s) rke2-ext-final-opensuse-rke2-ext-final-5bf455cf66-f4vzk: creating server [fleet-default/rke2-ext-final-opensuse-rke2-ext-final-36d034c5-f8bq8] of kind (HarvesterMachine) for machine rke2-ext-final-opensuse-rke2-ext-final-5bf455cf66-f4vzk in infrastructure provider, waiting for agent to check in and apply initial plan
 2022/05/05 00:20:52 [ERROR] error syncing '_all_': handler user-controllers-controller: failed to start user controllers for cluster c-m-2dbqq5sw: ClusterUnavailable 503: cluster not found, requeuing
 2022/05/05 00:20:54 [INFO] [node-controller-rancher-machine] The default lines below are for a sh/bash shell, you can specify the shell you're using, with the --shell flag.
 2022/05/05 00:20:54 [INFO] [node-controller-rancher-machine] 
 2022/05/05 00:20:59 [INFO] Generating and uploading node config opensuse-rke1-ext2
 2022/05/05 00:20:59 [ERROR] error syncing 'c-7m846/m-6wf9z': handler node-controller: Error creating machine: Error in driver during machine creation: Too many retries waiting for machine to be Running.  Last error: Maximum number of retries (120) exceeded, requeuing
 2022/05/05 00:22:52 [ERROR] error syncing '_all_': handler user-controllers-controller: failed to start user controllers for cluster c-m-2dbqq5sw: ClusterUnavailable 503: cluster not found, requeuing
 2022/05/05 00:22:59 [INFO] Removing node opensuse-rke1-ext2
 2022/05/05 00:22:59 [INFO] [node-controller-rancher-machine] About to remove opensuse-rke1-ext2
 2022/05/05 00:22:59 [INFO] [node-controller-rancher-machine] WARNING: This action will delete both local reference and remote instance.
 2022/05/05 00:22:59 [INFO] Creating jail for c-7m846
 2022/05/05 00:22:59 [INFO] Provisioning node opensuse-rke1-ext1
 2022/05/05 00:22:59 [INFO] [node-controller-rancher-machine] (opensuse-rke1-ext2) Remove node
 2022/05/05 00:22:59 [INFO] [node-controller-rancher-machine] Creating CA: /management-state/node/nodes/opensuse-rke1-ext1/certs/ca.pem
 2022/05/05 00:22:59 [INFO] [node-controller-rancher-machine] Creating client certificate: /management-state/node/nodes/opensuse-rke1-ext1/certs/cert.pem
 2022/05/05 00:22:59 [INFO] [node-controller-rancher-machine] (opensuse-rke1-ext2) Waiting for node removed
 2022/05/05 00:22:59 [INFO] [node-controller-rancher-machine] Running pre-create checks...
 2022/05/05 00:22:59 [INFO] [node-controller-rancher-machine] Creating machine...
 2022/05/05 00:22:59 [INFO] [node-controller-rancher-machine] (opensuse-rke1-ext1) Creating New SSH Key
 2022/05/05 00:23:00 [INFO] [node-controller-rancher-machine] (opensuse-rke1-ext1) Using public Key: /management-state/node/nodes/opensuse-rke1-ext1/machines/opensuse-rke1-ext1/id_rsa.pub
 2022/05/05 00:23:00 [INFO] [node-controller-rancher-machine] (opensuse-rke1-ext1) Start node
 2022/05/05 00:23:00 [INFO] [node-controller-rancher-machine] (opensuse-rke1-ext1) Waiting for node become Running
 2022/05/05 00:23:09 [INFO] [node-controller-rancher-machine] Successfully removed opensuse-rke1-ext2
 2022/05/05 00:23:09 [INFO] [node-controller-rancher-machine] (opensuse-rke1-ext2) Closing plugin on server side
 2022/05/05 00:23:09 [INFO] Removing node opensuse-rke1-ext2 done
 2022/05/05 00:23:09 [ERROR] error syncing 'c-7m846/m-6wf9z': handler node-controller: nodes.management.cattle.io "m-6wf9z" not found, requeuing
 2022/05/05 00:24:52 [ERROR] error syncing '_all_': handler user-controllers-controller: failed to start user controllers for cluster c-m-2dbqq5sw: ClusterUnavailable 503: cluster not found, requeuing

```

**Comment by joshmeranda** [comment]
This was an issues with authenticating the provisioning pod in [rancher/machine](https://github.com/rancher/machine) that has been resolved by `v0.15-rancher88`.

**Comment by guangbochen** [comment]
close as the external cloud credentials support will be deprecated via https://github.com/harvester/harvester/issues/3319.
